waiter/integration/waiter/grpc_test.clj

```diff
@@ -281,7 +281,33 @@
           (let [correlation-id (rand-name)
                 request-headers (assoc request-headers "x-cid" correlation-id)
                 grpc-client (initialize-grpc-client correlation-id host h2c-port)
-                rpc-result (.sendPackage grpc-client request-headers id from content 1000 10000)
+                sleep-ms 1000
+                deadline-ms 10000
+                rpc-result (.sendPackage grpc-client request-headers id from content sleep-ms deadline-ms)
+                ^CourierReply reply (.result rpc-result)
+                ^Status status (.status rpc-result)
+                assertion-message (grpc-result->assertion-message correlation-id rpc-result)]
+            (assert-grpc-ok-status status assertion-message)
+            (is reply assertion-message)
+            (when reply
+              (is (= id (.getId reply)) assertion-message)
+              (is (= content (.getMessage reply)) assertion-message)
+              (is (= "received" (.getResponse reply)) assertion-message))
+            (assert-request-state grpc-client request-headers service-id correlation-id ::success)))
+
+        (testing "long request and reply"
```

Comment:
This test fails without the fix with a `TimeoutException` in Waiter request to the backend.